pkg/service/spec.go

```diff
@@ -0,0 +1,74 @@
+package service
```

Comment:
Should we move this to `processor` package for the sake of import names in other packages?
Ex: `processor.Spec` instead of `service.Spec`?

Reply:
I was thinking about that then I realized that Spec is only being called by functions inside service package. So I feel that it belongs to service package. Wdyt?

Reply:
`Spec` on its own shouldn't be a problem since it is in the same package right now. I'm concerned about the cases when we import it in other packages. Also, it was named `ProcessSpec` before, so the imports looked like `service.ProcessSpec` which somewhat gave context to the type of spec we were referring to.
If `service.Spec` sounds okay in general, I'm okay with that too.

Reply:
Okay, I think we can change the struct to ProcessSpec but still keep the file name as `spec.go`.